File: 백준/Silver/2606. 바이러스/바이러스.py
from collections import deque

n = int(input())
m = int(input())
graph = [[] for _ in range(n+1)]
visited = [0]*(n+1)


for _ in range(m):
    x,y = map(int,input().split())
    graph[x].append(y)
    graph[y].append(x)


# print('graph는',graph)
def bfs():
    visited[1]=1
    queue = deque([1])
    while queue:
        v = queue.popleft()
        # print(v)
        for i in graph[v]:
            # print(visited)
            if not visited[i]:
                queue.append(i)
                visited[i] = 1

bfs()
count = 0
for num in visited:
    if num == 1:
        count+=1
    
print(count-1)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        int M = Integer.parseInt(br.readLine());

        List<List<Integer>> graph = new ArrayList<>();
        for (int i = 0; i <= N; i++) {
            graph.add(new ArrayList<>());
        }

        for (int i = 0; i < M; i++) {
            String[] input = br.readLine().split(" ");
            int x = Integer.parseInt(input[0]);
            int y = Integer.parseInt(input[1]);
            graph.get(x).add(y);
            graph.get(y).add(x);  // 양방향 그래프
        }

        boolean[] visited = new boolean[N+1];
        bfs(graph, visited);

        int count = 0;
        for (int i = 2; i <= N; i++) {  // 1번 컴퓨터를 제외하고 카운트
            if (visited[i]) count++;
        }

        System.out.println(count);
    }

    private static void bfs(List<List<Integer>> graph, boolean[] visited) {
        Queue<Integer> queue = new LinkedList<>();
        queue.offer(1);
        visited[1] = true;

        while (!queue.isEmpty()) {
            int v = queue.poll();
            for (int i : graph.get(v)) {
                if (!visited[i]) { //방문 확인 
                    queue.offer(i);
                    visited[i] = true;
                }
            }
        }
    }
}